File: frontend-admin/src/models/Agendamento.ts
import { Barbeiro } from "./Barbeiro";
import { Cliente } from "./Cliente";
import { FormaPagamento } from "./FormaPagamento";
import { Servico } from "./Servico";
import { Usuario } from "./Usuario";

export interface Agendamento {
    idExterno: number;
    dataHora: string;
    valorTotal: number;
    tempoDuracaoEmMinutos: number;
    cliente: Cliente;
    barbeiros: Barbeiro[];
    servicos: Servico[];
    formaPagamento: FormaPagamento;
    criadoPor: Usuario;
}

export interface AgendamentoReduzido {
    idExterno: string;
    nomeCliente: string;
    servicos: string[];
    dataHoraInicio: Date;
    dataHoraFim: Date;
}

export interface AgendamentoCadastro {
    dataHora: string;
    valorTotal: number;
    tempoDuracaoEmMinutos: number;
    barbeariaIdExterno: string;
    clienteIdExterno: string;
    servicosIdsExterno: string[];
    barbeirosIdsExterno: string[];
    formaPagamentoIdExterno: string;
}

File: backend/src/main/java/com/optimiza/clickbarber/model/agendamento/Agendamento.java
package com.optimiza.clickbarber.model.agendamento;

import com.optimiza.clickbarber.model.barbearia.Barbearia;
import com.optimiza.clickbarber.model.barbeiro.Barbeiro;
import com.optimiza.clickbarber.model.cliente.Cliente;
import com.optimiza.clickbarber.model.formaspagamento.FormaPagamento;
import com.optimiza.clickbarber.model.servico.Servico;
import com.optimiza.clickbarber.model.usuario.Usuario;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.ZonedDateTime;
import java.util.Set;
import java.util.UUID;

import static java.util.Objects.isNull;

@Table(name = "agendamentos")
@Entity
@Getter
@Setter
@ToString
@EqualsAndHashCode
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Agendamento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "id_externo", unique = true, nullable = false)
    private UUID idExterno;

    private ZonedDateTime dataHora;
    private BigDecimal valorTotal;
    private Integer tempoDuracaoEmMinutos;

    @ManyToOne
    @JoinColumn(name = "barbearia_id", nullable = false)
    private Barbearia barbearia;

    @ManyToOne
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente;

    @ManyToMany
    @JoinTable(
            name = "agendamentos_servicos",
            joinColumns = @JoinColumn(name = "agendamento_id"),
            inverseJoinColumns = @JoinColumn(name = "servico_id")
    )
    private Set<Servico> servicos;

    @ManyToMany
    @JoinTable(
            name = "agendamentos_barbeiros",
            joinColumns = @JoinColumn(name = "agendamento_id"),
            inverseJoinColumns = @JoinColumn(name = "barbeiro_id")
    )
    private Set<Barbeiro> barbeiros;

    @ManyToOne
    @JoinColumn(name = "forma_pagamento_id", nullable = false)
    private FormaPagamento formaPagamento;

    @ManyToOne
    @JoinColumn(name = "created_by", nullable = false)
    private Usuario createdBy;

    @PrePersist
    void gerarIdExterno() {
        if (isNull(idExterno)) {
            idExterno = UUID.randomUUID();
        }
    }

}
